src/modules/Embed/Embed.js

```diff
@@ -0,0 +1,198 @@
+import cx from 'classnames'
+import React, { PropTypes } from 'react'
+
+import {
+  AutoControlledComponent as Component,
+  customPropTypes,
+  getElementType,
+  getUnhandledProps,
+  META,
+  useKeyOnly,
+} from '../../lib'
+import Icon from '../../elements/Icon'
+
+const _meta = {
+  name: 'Embed',
+  type: META.TYPES.MODULE,
+  props: {
+    aspectRatio: ['4:3', '16:9', '21:9'],
+    source: ['youtube', 'vimeo'],
+  },
+}
+
+/**
+ * An embed displays content from other websites like YouTube videos or Google Maps.
+ */
+export default class Embed extends Component {
+  static autoControlledProps = [
+    'active',
+  ]
+
+  static defaultProps = {
+    icon: 'video play',
+  }
+
+  static _meta = _meta
+
+  static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** An embed can be active. */
+    active: PropTypes.bool,
+
+    /** An embed can specify an alternative aspect ratio. */
+    aspectRatio: PropTypes.oneOf(_meta.props.aspectRatio),
+
+    /** Primary content. */
+    children: PropTypes.node,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** Initial value of active. */
+    defaultActive: PropTypes.bool,
+
+    /** Specifies an icon to use with placeholder content. */
+    icon: customPropTypes.itemShorthand,
+
+    /** Сalled with event on Embed click with (event, props). */
+    onClick: PropTypes.func,
+
+    /** A placeholder image for embed. */
+    placeholder: PropTypes.string,
+
+    /** Specifies a source to use. */
+    source: customPropTypes.every([
+      customPropTypes.disallow(['sourceUrl']),
+      PropTypes.oneOf(_meta.props.source),
+    ]),
+
+    /** Setting to true or false will force autoplay. */
+    sourceAutoPlay: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.bool,
+    ]),
+
+    /** Whether to show networks branded UI like title cards, or after video calls to action. */
+    sourceBrandedUI: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.bool,
+    ]),
+
+    /** Specifies a default chrome color with Vimeo or YouTube. */
+    sourceColor: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.string,
+    ]),
+
+    /** Whether to show networks branded UI like title cards, or after video calls to action. */
+    sourceHd: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.bool,
+    ]),
+
+    /** Specifies an id for source. */
+    sourceId: customPropTypes.every([
+      customPropTypes.demand(['source']),
+      PropTypes.string,
+    ]),
+
+    /** Specifies a url to use for embed. */
+    sourceUrl: customPropTypes.every([
```

Comment:
One last update, can we remove the `source` prefixing from these props?  I'm going for parity with the Semantic-UI core settings:

![image](https://cloud.githubusercontent.com/assets/5067638/20185183/f022b97a-a71f-11e6-98b7-2384fa53322f.png)